Client/Frontend/Browser/URLBarView.swift

```diff
@@ -620,8 +620,8 @@ extension URLBarView: TabLocationViewDelegate {
 
     func tabLocationViewDidTapLocation(tabLocationView: TabLocationView) {
         var locationText = delegate?.urlBarDisplayTextForURL(locationView.url)
-        if let host = locationView.url?.host {
-            locationText = locationView.url?.absoluteString?.stringByReplacingOccurrencesOfString(host, withString: host.asciiHostToUTF8())
+        if let text = locationText, let url = NSURL(string: text), let host = url.host {
```

Comment:
Makes sense! Would you mind adding a comment in code explaining why we can't use `locationText.url` directly here?